### PR TITLE
Microchip MEC1501 / mec15xxevb_assy6853 board support

### PR DESCRIPTION
**Pull request by scottwcpg**
Microchip MEC1501 pull request made up of 4 commits:
1. Add MEC1501 header to ext/hal/microchip/mec/mec1501
2. Add MEC1501 DTSI file to dts/arm/microchip/
3. Add MEC1501 SoC folder and files to soc/arm/microchip_mec/
4. Add mec15xxevb_assy6853 board folder to boards/arm

### CODE REVIEW

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]:
```suggestion
   .. _MEC15xx EVB Schematic:
```

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]:
```suggestion
.. _SPI Dongle Schematic:
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Please see the doc comments on https://github.com/zephyrproject-rtos/zephyr/pull/15089 as most of them still apply to this PR.

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Need to add a link for this manual (at the end of the doc with all the others):
```
.. _MEC1501 Reference Manual:  <some URL here>
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
The PR looks ok, we are waiting for 1.14 to be release for this to go in. One main change that we need to address is the fact that the content in ext/ will need to go to a new and microchip related repo under the zephyr project GH org. I think we can merge this as is and do the split of ext/ components after the code is merged to avoid and conflicts or dependencies.

**Review by ghost — APPROVED** [review]

**Review by nashif — APPROVED** [review]
Can you add yourself to the CODEOWNERS file for the files you have added?

**Comment by nashif** [comment]
issues in https://github.com/zephyrproject-rtos/zephyr/pull/15101#issuecomment-478779066 needs to be addressed

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by andrewboie** [comment]
Need to fix some issues with licenses:

    ext/hal/microchip/mec/CMakeLists.txt missing license.
    soc/arm/microchip_mec/mec1501/CMakeLists.txt missing license.

Also one checkpatch.pl error needs to be addressed.
```

-:142: ERROR:TRAILING_WHITESPACE: trailing whitespace
#142: FILE: boards/arm/mec15xxevb_assy6853/doc/index.rst:45:
+- `MEC1501 Reference Manual`_ $
```

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: ``RAM`` and ``ROM``

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Better would be to join these lines into:
```
For more information about the SoC please see the `MEC1501 Reference Manual`_
```

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Better to join these lines as:
```
The default configuration can be found in the
:zephyr_file:`boards/arm/mec15xxevb_assy6853/mec15xxevb_assy6853_defconfig`
Kconfig file.

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]:
```suggestion
Peripheral Routing Jumpers
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (scottwcpg) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: Thanks. Changes in 1476db1

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (scottwcpg) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: Thanks. Done in 1476db16

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (scottwcpg) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: Thanks. Changed in 1476db1

**Comment by andrewboie** [comment]
Kconfig failing check needs to be addressed.

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Because of the double colons, I suspect you wanted this to appear as a "note", so add a blank line before this and change to:
```
.. note:: Remember that SPI MISO/MOSI are swapped on dediprog headers!
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
one typo, otherwise LGTM

File: boards/arm/mec15xxevb_assy6853/doc/index.rst
Comment (scottwcpg) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: Thanks! Pushed update with your recommended change.

**Review by dbkinder — APPROVED** [review]
+1 for docs, Thanks!

**Comment by andrewboie** [comment]
> Path '/arch/arm/include/cortex_m/cmse/' not found, in CODEOWNERS

this is a spurious failure. let me see what I can do

**Pull request merged**
